# **Analisis Performa Hero dalam kejuaraan M5 World Championship**

#**Import Library**

In [56]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

#**BUSINESS UNDERSTANDING**

##**Business Objective**

M5 World Championship adalah kejuaraan Mobile Legends Bang Bang (MLBB) tertinggi di dunia, yang sering dijadikan tolok ukur bagi para pemain MLBB. Analisis ini bertujuan untuk memberikan wawasan tentang performa hero di kejuaraan M5 agar para pemain MLBB dapat memilih hero dengan lebih baik.



##**Assess Situation**

Pertumbuhan pesat industri e-sports dan popularitas MLBB membuat para pemain semakin tertarik untuk memahami hero dan tren meta di level kompetisi tingkat dunia. Namun, kurangnya pemahaman yang mendalam tentang kinerja hero dan tren meta di M5 World Championship menjadi hambatan bagi para pemain dalam membuat keputusan strategis dalam permainan.

##**Data Mining Goals**

1. **Menganalisis Kinerja Hero:** Meneliti kinerja berbagai hero yang digunakan dalam M5 World Championship untuk mengidentifikasi hero yang efektif dan populer.

2. **Menganalisis Meta Permainan:** Memeriksa tren meta permainan yang dominan dalam M5 World Championship untuk memberikan wawasan tentang strategi dan gaya bermain yang efektif.

3. **Memberikan Rekomendasi kepada Pemain:** Memberikan rekomendasi kepada pemain MLBB tentang pilihan hero berdasarkan kinerja mereka di M5 World Championship.

4. **Membantu Pemain dalam Pengambilan Keputusan:** Membantu pemain MLBB dalam membuat keputusan strategis dalam pemilihan hero dan gaya bermain mereka dengan menyediakan informasi yang relevan tentang kinerja hero dan tren meta permainan.

##**Project Plan**

1. **Data Collection**: Mengumpulkan data kinerja hero dari sumber terpercaya, seperti statistik pick, win, dan loss.
2. **Data Preprocessing**: Membersihkan dan mempersiapkan data untuk analisis dengan menghilangkan entri yang tidak relevan atau tidak lengkap.
3. **Exploratory Data Analysis (EDA)**: Mengeksplorasi distribusi, tren, dan pola kinerja hero melalui visualisasi data.
4. **Advanced Analysis**: Melakukan analisis lebih lanjut untuk mengidentifikasi hero dominan dan tren strategi yang efektif.
5. **Visualization**: Membuat visualisasi untuk menyajikan hasil analisis dengan jelas dan mudah dimengerti.
6. **Documentation**: Mendokumentasikan metodologi, temuan, dan rekomendasi dalam sebuah laporan.
7. **Validation and Iteration**: Melakukan validasi hasil analisis dan iterasi tambahan jika diperlukan untuk memastikan kualitas dan keakuratan analisis.

#**DATA UNDERSTANDING**

##**Collect Initial Data**

Data ini berisi statistik hero yang digunakan di M5 World Championship
Sumber dari data ini kagle https://www.kaggle.com/datasets/myudak/m5-world-championship

In [57]:
df = pd.read_csv('M5_World_Championship.csv')
df.head()


Index        Hero  T_Picked  T_Wins  T_Loses T_WinRate T_PickPercentage  \
0      1   Valentina       727     334      393    45.94%           52.00%   
1      2    Fredrinn       726     383      343    52.75%           51.93%   
2      3     Terizla       620     348      272    56.13%           44.35%   
3      4       Edith       527     265      262    50.28%           37.70%   
4      5     Irithel       502     255      247    50.80%           35.91%   

   BS_Picked  BS_Wins  BS_Loses  ... RS_Loses  RS_WinRate  T_Banned  \
0        348      157       191  ...      202      46.70%       484   
1        361      184       177  ...      166      54.52%       475   
2        291      168       123  ...      149      54.71%       435   
3        258      125       133  ...      129      52.04%       477   
4        245      129       116  ...      131      49.03%       208   

   T_BansPercentage T_PicksBans  T_PicksBansPercentage  \
0            34.62%        1211                 86.62%   
1            33.98%        1201                 85.91%   
2            31.12%        1055                 75.46%   
3            34.12%        1004                 71.82%   
4            14.88%         710                 50.79%   

                                      Played By  \
0  19 (11-8)19 (10-9)19 (10-9)19 (7-12)17 (8-9)   
1  21 (15-6)18 (14-4)18 (12-6)18 (9-9)17 (10-7)   
2  22 (12-10)18 (10-8)15 (9-6)14 (10-4)14 (8-6)   
3     14 (9-5)14 (9-5)14 (6-8)13 (10-3)13 (9-4)   
4    16 (8-8)15 (10-5)15 (10-5)12 (7-5)12 (7-5)   

                                         Played With  \
0   164 (83-81) 162 (79-83) 140 (71-69) 121 (58-6...   
1   164 (83-81) 131 (81-50) 126 (69-57) 122 (69-5...   
2   162 (79-83) 114 (76-38) 108 (56-52) 105 (49-5...   
3   122 (69-53) 112 (48-64) 109 (55-54) 100 (42-5...   
4   140 (71-69) 126 (69-57) 108 (56-52) 100 (42-5...   

                                          Played Vs.     Roles  
0   218 (85-133) 178 (68-110) 174 (77-97) 173 (82...      Mage  
1   218 (133-85) 190 (106-84) 185 (79-106) 173 (8...   Support  
2   185 (106-79) 178 (110-68) 132 (70-62) 119 (65...   Fighter  
3   174 (97-77) 147 (76-71) 132 (62-70) 108 (51-5...      Tank  
4   145 (73-72) 132 (60-72) 120 (68-52) 110 (47-6...  Marksman  

[5 rows x 23 columns]

##**Describe Data**

####Informasi Dasar

In [58]:

jumlah_baris = df.shape[0]
jumlah_kolom = df.shape[1]

print("Jumlah baris:", jumlah_baris)
print("Jumlah kolom:", jumlah_kolom)

Jumlah baris: 115
Jumlah kolom: 23


Di dalam dataset ini terdapat 115 jumlah baris dan 23 jumlah kolom

####Informasi Lanjutan

In [59]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115 entries, 0 to 114
Data columns (total 23 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Index                  115 non-null    int64 
 1   Hero                   115 non-null    object
 2   T_Picked               115 non-null    int64 
 3   T_Wins                 115 non-null    int64 
 4   T_Loses                115 non-null    int64 
 5   T_WinRate              115 non-null    object
 6   T_PickPercentage       115 non-null    object
 7   BS_Picked              115 non-null    int64 
 8   BS_Wins                115 non-null    int64 
 9   BS_Loses               115 non-null    int64 
 10  BS_WinRate             115 non-null    object
 11  RS_Picked              115 non-null    int64 
 12  RS_Wins                115 non-null    int64 
 13  RS_Loses               115 non-null    int64 
 14  RS_WinRate             115 non-null    object
 15  T_Banned               



1. **Index**: Nomor indeks.
   - Jumlah: 115
   - Tipe Data: integer (int64)

2. **Hero**: Nama Hero.
   - Jumlah: 115
   - Tipe Data: object

3. **T_Picked**: Total Hero dipilih.
   - Jumlah: 115
   - Tipe Data: integer (int64)

4. **T_Wins**: Total kemenangan dengan Hero tersebut.
   - Jumlah: 115
   - Tipe Data: integer (int64)

5. **T_Loses**: Total kekalahan dengan Hero tersebut.
   - Jumlah: 115
   - Tipe Data: integer (int64)

6. **T_WinRate**: Persentase kemenangan (T_Wins / Total Games).
   - Jumlah: 115
   - Tipe Data: object

7. **T_PickPercentage**: Persentase pick (T_Picked / Total Games).
   - Jumlah: 115
   - Tipe Data: object

8. **BS_Picked**: Total Hero dipilih oleh Tim Biru.
   - Jumlah: 115
   - Tipe Data: integer (int64)

9. **BS_Wins**: Total kemenangan dengan Hero oleh Tim Biru.
   - Jumlah: 115
   - Tipe Data: integer (int64)

10. **BS_Loses**: Total kekalahan dengan Hero oleh Tim Biru.
    - Jumlah: 115
    - Tipe Data: integer (int64)

11. **BS_WinRate**: Persentase kemenangan oleh Tim Biru (BS_Wins / BS_Picked).
    - Jumlah: 115
    - Tipe Data: object

12. **RS_Picked**: Total Hero dipilih oleh Tim Merah.
    - Jumlah: 115
    - Tipe Data: integer (int64)

13. **RS_Wins**: Total kemenangan dengan Hero oleh Tim Merah.
    - Jumlah: 115
    - Tipe Data: integer (int64)

14. **RS_Loses**: Total kekalahan dengan Hero oleh Tim Merah.
    - Jumlah: 115
    - Tipe Data: integer (int64)

15. **RS_WinRate**: Persentase kemenangan oleh Tim Merah (RS_Wins / RS_Picked).
    - Jumlah: 115
    - Tipe Data: object

16. **T_Banned**: Jumlah kali Hero tersebut dibanned.
    - Jumlah: 115
    - Tipe Data: integer (int64)

17. **T_BansPercentage**: Persentase ban (T_Banned / Total Games).
    - Jumlah: 115
    - Tipe Data: object

18. **T_PicksBans**: Jumlah kali Hero tersebut dipilih atau dibanned.
    - Jumlah: 115
    - Tipe Data: integer (int64)

19. **T_PicksBansPercentage**: Persentase pick dan ban (Pick&Ban / Total Games).
    - Jumlah: 115
    - Tipe Data: object

20. **Played By**: Dimainkan oleh siapa.
    - Jumlah: 115
    - Tipe Data: object

21. **Played With**: Dimainkan bersama dengan siapa.
    - Jumlah: 115
    - Tipe Data: object

22. **Played Vs.**: Dimainkan melawan siapa.
    - Jumlah: 115
    - Tipe Data: object

23. **Roles**: Peran dari Hero.
    - Jumlah: 115
    - Tipe Data: object



####Informasi Statistik Deskriptif

In [60]:
df.describe()

Index    T_Picked      T_Wins     T_Loses   BS_Picked     BS_Wins  \
count  115.000000  115.000000  115.000000  115.000000  115.000000  115.000000   
mean    58.000000  121.521739   60.739130   60.782609   60.739130   30.043478   
std     33.341666  165.790946   84.578842   82.061250   82.204397   41.649412   
min      1.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%     29.500000    6.000000    3.000000    4.000000    3.000000    1.000000   
50%     58.000000   43.000000   25.000000   22.000000   20.000000   11.000000   
75%     86.500000  193.000000   89.000000  101.000000  101.500000   48.000000   
max    115.000000  727.000000  383.000000  393.000000  361.000000  184.000000   

         BS_Loses   RS_Picked     RS_Wins    RS_Loses    T_Banned  T_PicksBans  
count  115.000000  115.000000  115.000000  115.000000  115.000000   115.000000  
mean    30.695652   60.782609   30.695652   30.086957  121.513043   243.034783  
std     41.148676   84.206490   43.392572   41.381317  180.269358   322.978407  
min      0.000000    0.000000    0.000000    0.000000    0.000000     1.000000  
25%      2.000000    3.000000    1.000000    2.000000    2.500000    10.500000  
50%     12.000000   23.000000   12.000000   12.000000   24.000000    83.000000  
75%     51.500000   95.000000   44.500000   46.000000  174.500000   373.000000  
max    191.000000  379.000000  199.000000  202.000000  741.000000  1211.000000



1. **Index**:
   - Jumlah: 115
   - Rata-rata: 58.00
   - Standar deviasi: 33.34
   - Nilai minimum: 1.00
   - Kuartil bawah (25%): 29.50
   - Median (50%): 58.00
   - Kuartil atas (75%): 86.50
   - Nilai maksimum: 115.00
   - Tipe Data: integer (int64)

2. **T_Picked**: Total Hero dipilih.
   - Jumlah: 115
   - Rata-rata: 121.52
   - Standar deviasi: 165.79
   - Nilai minimum: 0.00
   - Kuartil bawah (25%): 6.00
   - Median (50%): 43.00
   - Kuartil atas (75%): 193.00
   - Nilai maksimum: 727.00
   - Tipe Data: integer (int64)

3. **T_Wins**: Total kemenangan dengan Hero tersebut.
   - Jumlah: 115
   - Rata-rata: 60.74
   - Standar deviasi: 84.58
   - Nilai minimum: 0.00
   - Kuartil bawah (25%): 3.00
   - Median (50%): 25.00
   - Kuartil atas (75%): 89.00
   - Nilai maksimum: 383.00
   - Tipe Data: integer (int64)

4. **T_Loses**: Total kekalahan dengan Hero tersebut.
   - Jumlah: 115
   - Rata-rata: 60.78
   - Standar deviasi: 82.06
   - Nilai minimum: 0.00
   - Kuartil bawah (25%): 4.00
   - Median (50%): 22.00
   - Kuartil atas (75%): 101.00
   - Nilai maksimum: 393.00
   - Tipe Data: integer (int64)

5. **BS_Picked**: Total Hero dipilih oleh Tim Biru.
   - Jumlah: 115
   - Rata-rata: 60.74
   - Standar deviasi: 82.20
   - Nilai minimum: 0.00
   - Kuartil bawah (25%): 3.00
   - Median (50%): 20.00
   - Kuartil atas (75%): 101.50
   - Nilai maksimum: 361.00
   - Tipe Data: integer (int64)

6. **BS_Wins**: Total kemenangan dengan Hero oleh Tim Biru.
   - Jumlah: 115
   - Rata-rata: 30.04
   - Standar deviasi: 41.65
   - Nilai minimum: 0.00
   - Kuartil bawah (25%): 1.00
   - Median (50%): 11.00
   - Kuartil atas (75%): 48.00
   - Nilai maksimum: 184.00
   - Tipe Data: integer (int64)

7. **BS_Loses**: Total kekalahan dengan Hero oleh Tim Biru.
   - Jumlah: 115
   - Rata-rata: 30.70
   - Standar deviasi: 41.15
   - Nilai minimum: 0.00
   - Kuartil bawah (25%): 2.00
   - Median (50%): 12.00
   - Kuartil atas (75%): 51.50
   - Nilai maksimum: 191.00
   - Tipe Data: integer (int64)

8. **RS_Picked**: Total Hero dipilih oleh Tim Merah.
   - Jumlah: 115
   - Rata-rata: 60.78
   - Standar deviasi: 84.21
   - Nilai minimum: 0.00
   - Kuartil bawah (25%): 3.00
   - Median (50%): 23.00
   - Kuartil atas (75%): 95.00
   - Nilai maksimum: 379.00
   - Tipe Data: integer (int64)

9. **RS_Wins**: Total kemenangan dengan Hero oleh Tim Merah.
   - Jumlah: 115
   - Rata-rata: 30.70
   - Standar deviasi: 43.39
   - Nilai minimum: 0.00
   - Kuartil bawah (25%): 1.00
   - Median (50%): 12.00
   - Kuartil atas (75%): 44.50
   - Nilai maksimum: 199.00
   - Tipe Data: integer (int64)

10. **RS_Loses**: Total kekalahan dengan Hero oleh Tim Merah.
    - Jumlah: 115
    - Rata-rata: 30.09
    - Standar deviasi: 41.38
    - Nilai minimum: 0.00
    - Kuartil bawah (25%): 2.00
    - Median (50%): 12.00
    - Kuartil atas (75%): 46.00
    - Nilai maksimum: 202.00
    - Tipe Data: integer (int64)

11. **T_Banned**: Jumlah kali Hero tersebut dibanned.
    - Jumlah: 115
    - Rata-rata: 121.51
    - Standar deviasi: 180.27
    - Nilai minimum: 0.00
    - Kuartil bawah (25%): 2.50
    - Median (50%): 24.00
    - Kuartil atas (75%): 174.50
    - Nilai maksimum: 741.00
    - Tipe Data: integer (int64)

12. **T_PicksBans**: Jumlah kali Hero tersebut dipilih atau dibanned.
    - Jumlah: 115
    - Rata-rata: 243.03
    - Standar deviasi: 322.98
    - Nilai minimum: 1.00
    - Kuartil bawah (25%): 10.50
    - Median (50%): 83.00
    - Kuartil atas (75%): 373.00
    - Nilai maksimum: 1211.00
    - Tipe Data: integer (int64)



Disini saya melakukan Data Cleaning agar datanya lebih mudah untuk di visualisasikan

In [61]:
df.drop(['Index', 'Played By', 'Played With','Played Vs.'], axis=1,inplace=True)
df.head()

Hero  T_Picked  T_Wins  T_Loses T_WinRate T_PickPercentage  \
0   Valentina       727     334      393    45.94%           52.00%   
1    Fredrinn       726     383      343    52.75%           51.93%   
2     Terizla       620     348      272    56.13%           44.35%   
3       Edith       527     265      262    50.28%           37.70%   
4     Irithel       502     255      247    50.80%           35.91%   

   BS_Picked  BS_Wins  BS_Loses BS_WinRate  RS_Picked  RS_Wins  RS_Loses  \
0        348      157       191     45.11%        379      177       202   
1        361      184       177     50.97%        365      199       166   
2        291      168       123     57.73%        329      180       149   
3        258      125       133     48.45%        269      140       129   
4        245      129       116     52.65%        257      126       131   

  RS_WinRate  T_Banned T_BansPercentage  T_PicksBans T_PicksBansPercentage  \
0     46.70%       484           34.62%         1211                86.62%   
1     54.52%       475           33.98%         1201                85.91%   
2     54.71%       435           31.12%         1055                75.46%   
3     52.04%       477           34.12%         1004                71.82%   
4     49.03%       208           14.88%          710                50.79%   

      Roles  
0      Mage  
1   Support  
2   Fighter  
3      Tank  
4  Marksman

In [62]:
df['T_WinRate'] = pd.to_numeric(df['T_WinRate'].str.strip('%'), errors='coerce')
df['T_PickPercentage'] = pd.to_numeric(df['T_PickPercentage'].str.strip('%'), errors='coerce')
df['BS_WinRate'] = pd.to_numeric(df['BS_WinRate'].str.strip('%'), errors='coerce')
df['RS_WinRate'] = pd.to_numeric(df['RS_WinRate'].str.strip('%'), errors='coerce')
df['T_BansPercentage'] = pd.to_numeric(df['T_BansPercentage'].str.strip('%'), errors='coerce')
df['T_PicksBansPercentage'] = pd.to_numeric(df['T_PicksBansPercentage'].str.strip('%'), errors='coerce')

##**Exploratory Data Analysis**

In [63]:
df_wr = df.sort_values('T_WinRate')

fig = px.bar(
    df_wr,
    x='Hero',
    y='T_WinRate',
    color='T_WinRate',
    labels={"T_WinRate":"Win Rate"},
    title='Distribusi Tingkat Win Rate Hero'
)
fig.show()

Bisa diliat pada visualisasi di atas tingkat kemenangan/Win rate dari hero yang di pick di M5 Miya,
 Moskov, Freya dan Saber memiliki tingkat kemenangan tertinggi dan Jawhead memiliki tingkat kemenangan terendah dan Tigreal, Silvana dan Zilong tidak memiliki win rate atau 0 win rate.

In [64]:
fig = px.bar(df.sort_values('T_WinRate'), x='Hero', y=['BS_WinRate', 'RS_WinRate'],
             title='Win Rates by Side',
             labels={'value': 'Win Rate', 'variable': 'Side'},
             color_discrete_map={'BS_WinRate': 'blue', 'RS_WinRate': 'red'},
             barmode='group')

fig.update_layout(xaxis_title='Hero', yaxis_title='Win Rate', legend_title='Side')
fig.show()



Bisa diliat dari visualisasi di atas menampilkan tingkat kemenangan/Win rate hero yang di pakai oleh tim biru/merah dimana hero-hero yang di pakai oleh tim merah memiliki win rate yang lebih tinggi.

In [65]:
role_counts = df['Roles'].value_counts().reset_index()
fig = px.bar(
    role_counts,
    x='Roles',
    y='index',
    labels={'index': 'Role', 'Roles': 'Roles'},
    title='Distribution of Hero Roles in the Dataset',
)
fig.update_layout(
    xaxis=dict(title='Roles'),
)
fig.show()


Dari visualisasi di atas, terlihat bahwa distribusi role-role hero pada M5 World Championship menunjukkan bahwa role yang paling banyak didistribusikan adalah mage, diikuti oleh fighter, dan role yang paling sedikit didistribusikan adalah marksman/mage, mage/support, mage/assassin, marksman/fighter, marksman/support, fighter/support, support/mage, support/assassin, dan mage/tank.

In [66]:
avg_win_rate_by_role = df.groupby('Roles')['T_WinRate'].mean().reset_index().sort_values(by='T_WinRate')

# Create a bar chart
fig = px.bar(avg_win_rate_by_role, x='Roles', y='T_WinRate',
             title='Average Win Rate by Hero Role',
             labels={'Roles': 'Hero Roles', 'T_WinRate': 'Average Win Rate'},
             color='T_WinRate',
             color_continuous_scale='Viridis')

fig.update_layout(xaxis_title='Hero Roles', yaxis_title='Average Win Rate')
fig.show()

Berdasarkan hasil visualisasi di atas kita dapat mengetahui role dengan rata-rata winrate tertinggi adalah Marksman/Mage dan role dengan rata-rata win rate terendah adalah Marksman/Fighter.

In [67]:
fig = px.bar(df.sort_values(by='T_Picked'), x='Hero', y='T_Picked',
             title='Overall Pick Percentage for Each Hero',
             labels={'Hero': 'Heroes', 'T_Picked': 'Pick Percentage'},
             color='T_Picked',
             color_continuous_scale='Viridis')

fig.update_layout(xaxis_title='Heroes', yaxis_title='Pick Percentage')
fig.show()

Dari hasil visualisasi diatas kit dapt mengetahui hero apa saja yang paling sering di pick dan yang paling sering di pick adalah hero valentina dan fredrin.

In [68]:
fig = px.scatter(df, x='T_Picked', y='T_WinRate',
                 title='Correlation between Hero Picks and Win Rates',
                 labels={'T_Picked': 'Total Heroes Picked', 'T_WinRate': 'Win Rate'},
                 color='Hero',
                 color_continuous_scale='Viridis')

fig.update_layout(xaxis_title='Total Heroes Picked', yaxis_title='Win Rate')
fig.show()

Dari visualisasi diatas kita dapat melihat korelasi dari total pick hero dengan win rate dari hero tersebut.

In [69]:
fig = px.bar(df.sort_values(by='T_BansPercentage'), x='Hero', y='T_BansPercentage',
             title='Overall Ban Percentage for Each Hero',
             labels={'Hero': 'Heroes', 'T_BansPercentage': 'Ban Percentage'},
             color='T_BansPercentage',
             color_continuous_scale='Viridis')

fig.update_layout(xaxis_title='Heroes', yaxis_title='Ban Percentage')
fig.show()

Dari visualisasi diatas kita dapat melihat hero hero apa saja yang sering di ban, dan angela adalah hero paling sering di ban pada kejuaraan M5.

In [70]:
fig = px.bar(df.sort_values(by='T_PicksBansPercentage'), x='Hero', y='T_PicksBansPercentage',
             title='Heroes with High Pick and Ban Percentages',
             labels={'Hero': 'Heroes', 'T_PicksBansPercentage': 'Pick and Ban Percentage'},
             color='T_PicksBansPercentage',
             color_continuous_scale='Viridis')

fig.update_layout(xaxis_title='Heroes', yaxis_title='Pick and Ban Percentage')
fig.show()

dari visualisasi diatas kita dapat melihat persentasi dari pick dan ban hero yang ada di kejuaraan M5 dimana presentasi pick dan ban tertinggi ad pada hero valentina dan disusul oleh hero fredrin.

In [71]:
fig = px.bar(df.sort_values(by="T_BansPercentage",ascending=False), x='Roles', y='T_BansPercentage',
             title='Ban Percentage by Hero Roles',
             labels={'Roles': 'Hero Roles', 'T_BansPercentage': 'Ban Percentage'},
             color='T_BansPercentage',
             color_continuous_scale='Viridis')

fig.show()

Berdasaran visualisasi diatas kita dapat melihat bahwa role hero dengan persentase ban yang paling tinggi adalah role mage.

##**Verify Data Quality**

####Missing Values

In [72]:
missing_values = df.isnull().sum()
print (missing_values)

Hero                      0
T_Picked                  0
T_Wins                    0
T_Loses                   0
T_WinRate                 2
T_PickPercentage          0
BS_Picked                 0
BS_Wins                   0
BS_Loses                  0
BS_WinRate               11
RS_Picked                 0
RS_Wins                   0
RS_Loses                  0
RS_WinRate                4
T_Banned                  0
T_BansPercentage          0
T_PicksBans               0
T_PicksBansPercentage     0
Roles                     0
dtype: int64



1. **Hero**: Tidak ada nilai yang hilang dalam kolom ini.
2. **T_Picked**: Tidak ada nilai yang hilang dalam kolom ini.
3. **T_Wins**: Tidak ada nilai yang hilang dalam kolom ini.
4. **T_Loses**: Tidak ada nilai yang hilang dalam kolom ini.
5. **T_WinRate**: Terdapat 2 nilai yang hilang dalam kolom ini.
6. **T_PickPercentage**: Tidak ada nilai yang hilang dalam kolom ini.
7. **BS_Picked**: Tidak ada nilai yang hilang dalam kolom ini.
8. **BS_Wins**: Tidak ada nilai yang hilang dalam kolom ini.
9. **BS_Loses**: Tidak ada nilai yang hilang dalam kolom ini.
10. **BS_WinRate**: Terdapat 11 nilai yang hilang dalam kolom ini.
11. **RS_Picked**: Tidak ada nilai yang hilang dalam kolom ini.
12. **RS_Wins**: Tidak ada nilai yang hilang dalam kolom ini.
13. **RS_Loses**: Tidak ada nilai yang hilang dalam kolom ini.
14. **RS_WinRate**: Terdapat 4 nilai yang hilang dalam kolom ini.
15. **T_Banned**: Tidak ada nilai yang hilang dalam kolom ini.
16. **T_BansPercentage**: Tidak ada nilai yang hilang dalam kolom ini.
17. **T_PicksBans**: Tidak ada nilai yang hilang dalam kolom ini.
18. **T_PicksBansPercentage**: Tidak ada nilai yang hilang dalam kolom ini.
19. **Roles**: Tidak ada nilai yang hilang dalam kolom ini.


####Outliers Values

In [73]:
fig = px.box(df, y='T_WinRate', title='Box Plot of Overall Win Rates')
fig.update_layout(xaxis_title='Win Rate')
fig.show()

Dari visualisasi diatas kita dapat mengetahui bahwa dalam keseluruhan game yang total win ratenya 100, 80, dan 0 sangat jarang.

####Duplicated Values

In [74]:
print("Total Duplication: ", df.duplicated().sum())

Total Duplication:  0


Berdasarkan hasil diatas kita dapat mengetahui bahwa tidak ada data yang terduplikasi

####Inconsistent Values / Noise (Optional)

In [75]:
#Tambahkan kode untuk mengecek apakah ada nilai inkosisten/noise pada tiap kolom di dataset kalian ya (OPSIONAL)

Jangan lupa juga untuk berikan penjelasan di sini (OPSIONAL)